spec/requests/jobseekers/job_applications_spec.rb

```diff
@@ -35,6 +56,81 @@
     end
   end
 
+  describe "POST #submit" do
+    let!(:job_application) { create(:job_application, jobseeker: jobseeker, vacancy: vacancy) }
+    let(:button) { I18n.t("buttons.submit_application") }
+    let(:confirm_data_accurate) { 1 }
+    let(:confirm_data_usage) { 1 }
+    let(:params) do
+      {
+        jobseekers_job_application_review_form:
+          { confirm_data_accurate: confirm_data_accurate, confirm_data_usage: confirm_data_usage },
+        commit: button,
+      }
+    end
+
+    context "when the job is not listed" do
+      let(:vacancy) { create(:vacancy, :expired, organisation_vacancies_attributes: [{ organisation: build(:school) }]) }
+
+      it "raises an error and does not submit the job application or send email" do
+        assert_emails 0 do
+          expect { post jobseekers_job_application_submit_path(job_application.id), params: params }
+            .to not_change { job_application.reload.status }
+            .and raise_error(ActionController::RoutingError, /non-listed/)
+        end
+      end
+    end
+
+    context "when the job application is not a draft" do
+      let!(:job_application) { create(:job_application, :status_submitted, jobseeker: jobseeker, vacancy: vacancy) }
+
+      it "raises an error and does not submit the job application or send email" do
+        assert_emails 0 do
+          expect { post jobseekers_job_application_submit_path(job_application.id), params: params }
+            .to not_change { job_application.reload.status }
+            .and raise_error(ActionController::RoutingError, /non-draft/)
+        end
+      end
+    end
+
+    context "when the commit is `Save as draft`" do
```

Comment:
personally i find this wording a bit confusing as to what a 'commit' is and hence what this describes the test is supposed to do

Reply:
I added the word `param` which should clarify things